week-3/Homework/mandatory/4-eligible-students.js

```diff
@@ -3,13 +3,51 @@
 
   Create a function which:
   - Accepts an array which contains all the students' names and their attendance counts
-    (see tests to confirm how this data will be structured)
-  - Returns an array containing only the names of the who have attended AT LEAST 8 classes
-*/
+  (see tests to confirm how this data will be structured)
+  - Returns an array containing only the names of the who have attended AT LEAST 8 classes and plus
 
-function eligibleStudents() {
+  oldArr
+  let newArr
 
+  const attendances = [
+  ["Ahmed", 8],
+  ["Clement", 10],
+  ["Elamin", 6], 
+  ["Adam", 7],
+  ["Tayoa", 11],
+  ["Nina", 10]
+]
+- sort() in descending 
+- filter() - returns every element 
+- forEach()
+- multi-dimentional array sort
+
+*/ //[[string index 0],[number =index 1]]
+
+// function eligibleStudents(attendance) {
+//   let studentNames2 = attendance.filter(e => e[1] >= 8)
+//   return studentNames2.map(e => e[0])
+// }
+
+function eligibleStudents(oldStudentlist) {
+  let newStudentlist = [];
+  //loops through each student name
+for(let i = 0; i < oldStudentlist.length; ++i) {
```

Comment:
This works well :) You may also want to try writing this with a `filter` and a `map` to practice, like you did in the journey planner :)